doc/src/manual/parallel-computing.md

```diff
@@ -1,10 +1,265 @@
+returning from the function. The feeder tasks are able to share state via `nextidx` because
+they all run on the same process. No locking is required, since the threads are scheduled cooperatively
+and not preemptively. This means context switches only occur at well-defined points: in this case,
+when [`remotecall_fetch`](@ref) is called.
+
+## Channels
+
+The section on [`Task`](@ref)s in [Control Flow](@ref) discussed the execution of multiple functions in
+a co-operative manner. [`Channel`](@ref)s can be quite useful to pass data between running tasks, particularly
+those involving I/O operations.
+
+Examples of operations involving I/O include reading/writing to files, accessing web services,
+executing external programs, etc. In all these cases, overall execution time can be improved if
+other tasks can be run while a file is being read, or while waiting for an external service/program
+to complete.
+
+A channel can be visualized as a pipe, i.e., it has a write end and read end.
+
+  * Multiple writers in different tasks can write to the same channel concurrently via [`put!`](@ref)
+    calls.
+  * Multiple readers in different tasks can read data concurrently via [`take!`](@ref) calls.
+  * As an example:
+
+    ```julia
+    # Given Channels c1 and c2,
+    c1 = Channel(32)
+    c2 = Channel(32)
+
+    # and a function `foo` which reads items from from c1, processes the item read
+    # and writes a result to c2,
+    function foo()
+        while true
+            data = take!(c1)
+            [...]               # process data
+            put!(c2, result)    # write out result
+        end
+    end
+
+    # we can schedule `n` instances of `foo` to be active concurrently.
+    for _ in 1:n
+        @schedule foo()
+    end
+    ```
+  * Channels are created via the `Channel{T}(sz)` constructor. The channel will only hold objects
+    of type `T`. If the type is not specified, the channel can hold objects of any type. `sz` refers
+    to the maximum number of elements that can be held in the channel at any time. For example, `Channel(32)`
+    creates a channel that can hold a maximum of 32 objects of any type. A `Channel{MyType}(64)` can
+    hold up to 64 objects of `MyType` at any time.
+  * If a [`Channel`](@ref) is empty, readers (on a [`take!`](@ref) call) will block until data is available.
+  * If a [`Channel`](@ref) is full, writers (on a [`put!`](@ref) call) will block until space becomes available.
+  * [`isready`](@ref) tests for the presence of any object in the channel, while [`wait`](@ref)
+    waits for an object to become available.
+  * A [`Channel`](@ref) is in an open state initially. This means that it can be read from and written to
+    freely via [`take!`](@ref) and [`put!`](@ref) calls. [`close`](@ref) closes a [`Channel`](@ref).
+    On a closed [`Channel`](@ref), [`put!`](@ref) will fail. For example:
+
+```julia-repl
```

Comment:
Does this need to be indented?

Reply:
I left it indented to keep it in line with rest of the paragrah, but yeah maybe it is better to align it on the read line. What if I de-indent the text and indent the code example below?
Yeah maybe making a dozen of points after a paragraph is an excessive digression...